src/__tests__/dialog.spec.tsx

```diff
@@ -0,0 +1,176 @@
+                fireEvent.change(inputBox, { target: { value: DEFAULT_INVALID_API_KEY } })
+                const confirmButton = screen.getByText('CONFIRM', { selector: 'button' })
+                fireEvent.click(confirmButton)
+            })
+            await waitFor(() => {
+                expect(screen.getByLabelText('Please enter valid API Key')).toBeVisible()
+            })
+        })
+
+        /* 
+        Test Flow: 
+        1. When the user inputs valid API Key
+        2. We will see the deploy button in next (assume the user)
+        */
+        test("Test Valid API Key Input Will Move To Deploy Stage", async () => {
+            mockPostAPI("success", DEFAULT_VALID_API_KEY)
+            renderDialogContent();
+            await waitFor(() => {
+                const inputBox = screen.getByLabelText('API Key')
+                fireEvent.change(inputBox, { target: { value: DEFAULT_VALID_API_KEY } })
+                const confirmButton = screen.getByText('CONFIRM', { selector: 'button' })
+                fireEvent.click(confirmButton)
+            })
+            expect(await screen.findByText('Clicking on deploy will upload your notebook to Ploomber Cloud servers')).toBeVisible()
+        })
+    })
+
+    describe("API Deploy Stage", () => {
+
+        /* Before:
+            1. Valid API Key saved in user's environment
+        */
+        beforeEach(() => {
+            jest.clearAllMocks();
+            mockEnvWithAPI(DEFAULT_VALID_API_KEY);
+        });
+
+        /*
+        Ensure layout components render properly
+        */
+        test("Test First Time Deployment Layout", async () => {
+            renderDialogContent();
+
+            expect(await screen.findByText('Clicking on deploy will upload your notebook to Ploomber Cloud servers')).toBeVisible()
+            expect(await screen.getByRole('button')).toHaveTextContent('CONFIRM')
+
+        })
+        /* 
+        Test Flow: When the notebook have not been deployed before
+        */
+        test("Test First Time Deployment Click Confim", async () => {
+            mockJobDeployResult();
+            renderDialogContent();
+            expect(await screen.findByText('Clicking on deploy will upload your notebook to Ploomber Cloud servers')).toBeVisible()
+            expect(await screen.getByRole('button')).toHaveTextContent('CONFIRM')
+
+            const confirmButton = screen.getByText('CONFIRM', { selector: 'button' })
+            fireEvent.click(confirmButton)
+
+            expect(await screen.findByText("Check your deployment status here:")).toBeVisible()
+            expect(await screen.findByText(`https://platform.ploomber.io/dashboards/${DEFAULT_PROJECT_ID}/${DEFAULT_JOB_ID}`)).toBeVisible()
+        })
+        /* 
+        Test Flow: When the notebook have been deployed before
+        */
+        // test("Test Existing Project Re-deployment Success", async () => {
+        //     mockJobDeployResult(DEFAULT_PROJECT_ID);
```

Comment:
either leave this as a test or remove the code

Reply:
addressed